Add tests and update test harness.

jpco opened the pull request:
In addition to what's easily visible here, this change grows the "really long word" in trip.es because the buffer it's trying to overflow was changed from 1000 to 2048 bytes a while back.  I made sure to keep it Lewis Carroll, though. :)

wryun reviewed (approved):
:+1: maybe it's worth removing that 'bad' character from trip.es...

wryun commented on test/tests/option.es:
```diff
@@ -0,0 +1,59 @@
+		rm -f $temp
+	}
+
+	# TODO: test #119
```
A little cryptic to me...

jpco commented:
Removed the NUL byte with 'printf'.  _rc_ uses a dedicated "tripping" binary but as far as I can tell printf (and its `%b` flag) are portable POSIX at this point so it seems very reasonable to use.

I also removed the cryptic TODO and replaced it with a real test case.
